classes/Hooks/DisplayBackOfficeHeader.php

```diff
@@ -0,0 +1,249 @@
+            !$this->updateNotificationConfiguration->getTimestamp()
+            || (time() - $this->updateNotificationConfiguration->getTimestamp()) > self::INTERVAL_CHECK_TIME_IN_SECONDS
+        ) {
+            $this->checkNewerVersion();
+        }
+
+        $this->addScriptsVariables();
+
+        $request = Request::createFromGlobals();
+        $this->addUIAssets($request);
+
+        if (!$this->isEmployeeDefaultController()) {
+            return $this->content;
+        }
+
+        $employees = $this->updateNotificationConfiguration->getEmployees();
+
+        $employeeExists = array_filter($employees, function ($employee) {
+            return $employee['employeeID'] === $this->employeeId;
+        });
+
+        if ((empty($employeeExists) || time() > $employeeExists[0]['timestamp']) && $this->updateIsAvailable()) {
+            $this->content .= $this->container->getTwig()->render('@ModuleAutoUpgrade/hooks/external-layout.html.twig', $this->getParams());
+        }
+
+        return $this->content;
+    }
+
+    /**
+     * @throws LoaderError
+     * @throws RuntimeError
+     * @throws SyntaxError
+     */
+    private function addScriptsVariables(): void
+    {
+        $adminDir = trim(str_replace(_PS_ROOT_DIR_, '', realpath(_PS_ADMIN_DIR_)), DIRECTORY_SEPARATOR);
+
+        $scriptsVariables = [
+            'token' => Tools::getAdminTokenLite('AdminAutoupgradeAjax'),
+            'admin_url' => __PS_BASE_URI__ . $adminDir,
+            'admin_dir' => $adminDir,
+        ];
+
+        $this->content .= $this->container->getTwig()->render('@ModuleAutoUpgrade/module-script-variables.html.twig', [
+            'autoupgrade_variables' => $scriptsVariables,
+        ]);
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @throws LoaderError
+     * @throws RuntimeError
+     * @throws SyntaxError
+     */
+    private function addUIAssets(Request $request): void
+    {
+        $assetsEnvironment = $this->container->getAssetsEnvironment();
+        $assetsBaseUrl = $assetsEnvironment->getAssetsBaseUrl($request);
+        $twig = $this->container->getTwig();
+
+        if ($assetsEnvironment->isDevMode()) {
+            $this->context->controller->addCSS($assetsBaseUrl . '/src/scss/appUpdateNotification/main.scss');
+            $this->content .= $twig->render('@ModuleAutoUpgrade/module-script-tag.html.twig', ['module_type' => true, 'src' => $assetsBaseUrl . '/src/ts/appUpdateNotification/main.ts']);
+        } else {
+            $this->context->controller->addCSS($assetsBaseUrl . '/css/autoupgrade-notification.css');
+            $this->content .= $twig->render('@ModuleAutoUpgrade/module-script-tag.html.twig', ['module_type' => true, 'src' => $assetsBaseUrl . '/js/autoupgrade-notification.js?version=' . $this->psVersion]);
+        }
+    }
+
+    private function isEmployeeDefaultController(): bool
+    {
+        $controller = Tools::getValue('controller');
+        $employee = new Employee($this->employeeId);
```

Comment:
```suggestion
        $employee = $this->context->employee;
```

This can save us from a request